src/script_lua.c

```diff
@@ -824,17 +848,42 @@ static robj **luaArgsToRedisArgv(lua_State *lua, int *argc) {
      * is not a string or an integer (lua_isstring() return true for
      * integers as well). */
     if (j != *argc) {
-        j--;
-        while (j >= 0) {
-            decrRefCount(argv[j]);
-            j--;
-        }
-        zfree(argv);
+        freeLuaRedisArgv(lua_argv, &j);
         luaPushError(lua, "Lua redis lib command arguments must be strings or integers");
         return NULL;
     }
 
-    return argv;
+    return lua_argv;
+}
+
+void freeLuaRedisArgv(robj **argv, int* argc) {
```

Comment:
it's no longer updated.
```suggestion
void freeLuaRedisArgv(robj **argv, int argc) {
```

Reply:
👍  @sundb. Fixed in the latest commit across all usage